src/metacheck.cpp

```diff
@@ -147,7 +147,7 @@ void metacheck(TypedAST::Declaration* ast, TypeChecker& tc) {
 
 	if (ast->m_type_hint) {
 		metacheck(ast->m_type_hint, tc);
-		assert(ast->m_type_hint->m_meta_type != tc.meta_value() && "Type hint must not be a value");
+		assert(ast->m_type_hint->m_meta_type == tc.meta_monotype() && "Type hint must be a monotype");
```

Comment:
I'm not sure... maybe we should unify instead of check?

Reply:
Fair point. I still like to have an assert there for us when we debug and test.